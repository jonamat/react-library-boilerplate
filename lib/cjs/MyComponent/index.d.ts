import { FC, ReactNode } from 'react';
interface MyComponentProps {
    /** Inline prop doc example. Default `world` */
    myProp?: ReactNode;
}
/** MyComponent description. See [Docs](https://github.com/jonamat/react-library-boilerplate). */
declare const MyComponent: FC<MyComponentProps>;
export default MyComponent;
