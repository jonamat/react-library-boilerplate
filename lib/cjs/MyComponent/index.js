"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
/** MyComponent description. See [Docs](https://github.com/jonamat/react-library-boilerplate). */
var MyComponent = function (_a) {
    var _b = _a.myProp, myProp = _b === void 0 ? 'world' : _b;
    return react_1.default.createElement(react_1.default.Fragment, null,
        "Hello ",
        myProp,
        "!");
};
MyComponent.propTypes = {
    myProp: prop_types_1.default.element,
};
exports.default = MyComponent;
