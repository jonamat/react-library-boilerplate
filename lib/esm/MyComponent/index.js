import React from 'react';
import PropTypes from 'prop-types';
/** MyComponent description. See [Docs](https://github.com/jonamat/react-library-boilerplate). */
const MyComponent = ({ myProp = 'world' }) => {
    return React.createElement(React.Fragment, null,
        "Hello ",
        myProp,
        "!");
};
MyComponent.propTypes = {
    myProp: PropTypes.element,
};
export default MyComponent;
