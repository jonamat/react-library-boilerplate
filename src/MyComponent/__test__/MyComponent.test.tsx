import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { create } from 'react-test-renderer';

import MyComponent from '..';

const consoleError = jest.fn();
Object.defineProperty(console, 'error', {
    get() {
        return consoleError;
    },
});

beforeEach(() => {
    consoleError.mockClear();
});

const renderComponent = (props?: React.ComponentProps<typeof MyComponent>) => render(<MyComponent {...props} />);

describe('Component', () => {
    it('render properly', () => {
        expect(() => {
            const { queryByTestId } = renderComponent({ myProp: <span data-testid="children">mars</span> });

            expect(consoleError).not.toBeCalled();
            expect(queryByTestId('children')).toBeInTheDocument();
        }).not.toThrowError();
    });

    it('validates props', () => {
        expect(() => renderComponent({ myProp: { not: 'a react node' } })).toThrowError();
    });

    it('matches the snapshot', () => {
        const componentSnapshoot = create(<MyComponent />).toJSON();
        expect(componentSnapshoot).toMatchSnapshot();
    });
});
