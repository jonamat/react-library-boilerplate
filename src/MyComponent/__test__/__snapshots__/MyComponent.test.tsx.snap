// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches the snapshot 1`] = `
Array [
  "Hello ",
  "world",
  "!",
]
`;

exports[`Component matches the snapshot 2`] = `
Array [
  "Hello ",
  "world",
  "!",
]
`;
